forum/models.py

```diff
@@ -112,10 +115,31 @@ def __unicode__(self):
         return self.title
 
 
+@receiver(post_save, sender=Thread)
+def update_num_threads_on_thread_insert(**kwargs):
+    thread = kwargs['instance']
+    if kwargs['created']:
+        thread.forum.num_threads = F('num_threads') + 1
+        thread.forum.save()
+
+
+@receiver(pre_save, sender=Thread)
+def update_num_threads_on_thread_update(sender, instance, **kwargs):
+    if instance.id:
+        old_thread = Thread.objects.get(pk=instance.id)
+        if old_thread.forum_id != instance.forum_id:
+            old_thread.forum.num_threads = F('num_threads') - 1
+            old_thread.forum.save()
+            instance.forum.num_threads = F('num_threads') + 1
+            instance.forum.save()
+
+
 @receiver(post_delete, sender=Thread)
 def update_last_post_on_thread_delete(**kwargs):
     thread = kwargs['instance']
     try:
+        thread.forum.refresh_from_db()
```

Comment:
Transaction for this block.
There's a little bit of hidden magic here, where `set_last_post` performs the `.save()` that applies the update to `num_threads`. 
You have a similar comment at https://github.com/MTG/freesound/pull/968/files#diff-c1b18b9ac76c85da21bb14f4e0481a14R181, Do you think that we can improve this? Possible options:
 * @ffont has `commit=True` default parameters in some methods (https://github.com/MTG/freesound/blob/936b5d233fde0be045f48e40ee10793f166ae27e/sounds/models.py#L651) which allows us to force or ignore this and call it again separately
 * rename the method to not be `set_` so that it makes it clear that this sets and saves the method?
 * Documentation in `set_last_post` to say that it saves the object and that this behaviour is used elsewhere in the code?
 * Remove the save and require it to be added after any call to the method